GoodThingsComein3s/Views/RestaurantTableViewCell.m

```diff
@@ -21,9 +45,31 @@ -(void) setRestaurant:(Restaurant *)restaurant{
         self.restaurantPriceLabel.text = self.restaurant.price;
         self.restaurantImageView.image = self.restaurant.restaurantImage;
         self.ratingImageView.image = self.restaurant.ratingImage;
+        [self.likeButton setImage:[UIImage systemImageNamed:@"heart"] forState:UIControlStateNormal];
+        
+        if([PFUser currentUser] != nil){
+            for( PFObject *likedRestaurant in [PFUser currentUser][@"likedRestaurants"]){
+                [likedRestaurant fetchIfNeeded];
+                if([likedRestaurant[@"yelpID"] isEqualToString:self.restaurant.restaurantID]){
+                    [self.likeButton setImage:[UIImage systemImageNamed:@"heart.fill"] forState:UIControlStateNormal];
+                }
+            }
+        }
+        
+        
     }
     
     
 }
 
+- (PFObject *)restaurantToParseObject {
```

Comment:
Nice! This is serializing the object into a form that can be stored in the database